sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -2512,31 +2512,151 @@ def hpat_pandas_series_add(self, other, level=None, fill_value=None, axis=0):
     """
 
     _func_name = 'Method add().'
-    ty_checker = TypeChecker(_func_name)
-    ty_checker.check(self, SeriesType)
+    ty_checker = TypeChecker('Operator add().')
+    self_is_series, other_is_series = isinstance(self, SeriesType), isinstance(other, SeriesType)
+    if not (self_is_series or other_is_series):
+        return None
 
-    if not (isinstance(fill_value, types.Omitted) or fill_value is None):
-        ty_checker.raise_exc(fill_value, 'None', 'fill_value')
+    # this overload is not for string series
+    self_is_string_series = self_is_series and isinstance(self.dtype, types.UnicodeType)
+    other_is_string_series = other_is_series and isinstance(other.dtype, types.UnicodeType)
+    if self_is_string_series or other_is_string_series:
+        return None
 
-    if not (isinstance(axis, types.Omitted) or axis == 0):
+    if not isinstance(self, (SeriesType, types.Number)):
+        ty_checker.raise_exc(self, 'pandas.series or scalar', 'self')
+
+    if not isinstance(other, (SeriesType, types.Number)):
+        ty_checker.raise_exc(other, 'pandas.series or scalar', 'other')
+
+    operands_are_series = self_is_series and other_is_series
+    if operands_are_series:
+        none_or_numeric_indexes = ((isinstance(self.index, types.NoneType) or check_index_is_numeric(self))
+                                   and (isinstance(other.index, types.NoneType) or check_index_is_numeric(other)))
+        series_indexes_comparable = check_types_comparable(self.index, other.index) or none_or_numeric_indexes
+        if not series_indexes_comparable:
+            raise TypingError('{} Not implemented for series with not-comparable indexes. \
+            Given: self.index={}, other.index={}'.format(_func_name, self.index, other.index))
+
+    series_data_comparable = check_types_comparable(self, other)
+    if not series_data_comparable:
+        raise TypingError('{} Not supported for not-comparable operands. \
+        Given: self={}, other={}'.format(_func_name, self, other))
+
+    if not isinstance(level, types.Omitted) and level is not None:
+        ty_checker.raise_exc(level, 'None', 'level')
+
+    if not isinstance(fill_value, (types.Omitted, types.Number, types.NoneType)) and fill_value is not None:
+        ty_checker.raise_exc(fill_value, 'number', 'fill_value')
+
+    if not isinstance(axis, types.Omitted) and axis != 0:
         ty_checker.raise_exc(axis, 'int', 'axis')
+    fill_value_is_none = False
+    if isinstance(fill_value, (types.NoneType, types.Omitted)) or fill_value is None:
+        fill_value_is_none = True
```

Comment:
Maybe just:
`fill_value_is_none = isinstance(fill_value, (types.NoneType, types.Omitted)) or fill_value is None`
?